sentinel/pkg/checker/dal/app.go

```diff
@@ -110,6 +163,98 @@ func checkDal(endpoint string, key string, alarmCount map[string]int) error {
 	return nil
 }
 
+func checkDalWs(ctx context.Context) {
+	log.Debug().Msg("checking msg delays")
+	msgs := extractWsAlarms(ctx)
+	if len(msgs) > 0 {
+		alert.SlackAlert(strings.Join(msgs, "\n"))
+	}
+
+	log.Debug().Msg("checking msg push")
+	msgsNotRecieved := []string{}
+	updateTimes.Range(func(key, value interface{}) bool {
+		if recievedTime, ok := value.(time.Time); ok {
+			diff := time.Since(recievedTime)
+			if diff > 1*time.Second {
+				symbol := key.(string)
+				msg := fmt.Sprintf("(%s) ws not pushed for %v(sec)", symbol, diff.Seconds())
+				msgsNotRecieved = append(msgsNotRecieved, msg)
+			}
+		}
+		return true
+	})
+	if len(msgsNotRecieved) > 0 {
+		alert.SlackAlert(strings.Join(msgsNotRecieved, "\n"))
+	}
+}
+
+func extractWsAlarms(ctx context.Context) []string {
+	var msgs = []string{}
+	select {
+	case <-ctx.Done():
+		return nil
+	case entry := <-wsMsgChan:
+		msgs = append(msgs, entry)
+	loop:
+		for {
+			select {
+			case entry := <-wsMsgChan:
+				msgs = append(msgs, entry)
+			default:
+				break loop
+			}
+		}
+	}
+	return msgs
+}
```

Comment:
I think the function will get stuck if there is no entry from both `ctx` and `wsMsgChan`, because the current select design is blocking. If that's intentional, please, ignore this comment

Reply:
awh I see, will make an update